Extension/src/LanguageServer/client.ts

```diff
@@ -674,47 +689,63 @@ export class DefaultClient implements Client {
-                                this.client.notifyWhenReady(() => {
-                                    // The current request is represented by referencesParams.  If a request detects
-                                    // referencesParams does not match the object used when creating the request, abort it.
-                                    if (params !== referencesParams) {
-                                        reject();
-                                    }
-                                    referencesPromiseReject = reject;
-                                    referencesRequestPending = true;
-                                    this.client.languageClient.sendNotification(RenameNotification, params);
-
-                                    this.client.references.setResultsCallback(referencesResult => {
-                                        referencesRequestPending = false;
-                                        renamePending = false;
-                                        let workspaceEdit: vscode.WorkspaceEdit = new vscode.WorkspaceEdit();
-                                        // If rename UI Was cancelled, we will get a null result
-                                        // If null, return an empty list to avoid Rename failure dialog
-                                        if (referencesResult !== null) {
-                                            for (let reference of referencesResult.referenceInfos) {
-                                                let uri: vscode.Uri = vscode.Uri.file(reference.file);
-                                                let range: vscode.Range = new vscode.Range(reference.position.line, reference.position.character, reference.position.line, reference.position.character + referencesResult.text.length);
-                                                workspaceEdit.replace(uri, range, newName);
-                                            }
+                                let callback: () => void = () => {
+                                    let params: RenameParams = {
+                                        newName: newName,
+                                        position: Position.create(position.line, position.character),
+                                        textDocument: this.client.languageClient.code2ProtocolConverter.asTextDocumentIdentifier(document)
+                                    };
+                                    referencesParams = params;
+                                    this.client.notifyWhenReady(() => {
+                                        // The current request is represented by referencesParams.  If a request detects
+                                        // referencesParams does not match the object used when creating the request, abort it.
+                                        if (params !== referencesParams) {
+                                            reject();
```

Comment:
`reject` does not return.  So unless you want to run the rest of the code in this function after rejecting, you should early return here.

Reply:
I do want it to run the rest of the code in that function.

Reply:
Ack, you were correct, this one should have returned.